internal/server/data/group_test.go

```diff
@@ -83,17 +86,57 @@ func TestListGroups(t *testing.T) {
 		product   = models.Group{Name: "Product"}
 	)
 
-	createGroups(t, db, everyone, engineers, product)
+	createGroups(t, db, &everyone, &engineers, &product)
 
-	groups, err := ListGroups(db)
-	assert.NilError(t, err)
-	assert.Equal(t, 3, len(groups))
+	firstUser := models.Identity{
+		Name:   "firstly",
+		Groups: []models.Group{everyone, engineers},
+	}
+	secondUser := models.Identity{
+		Name:   "secondarly",
+		Groups: []models.Group{everyone, product},
+	}
+	createIdentities(t, db, &firstUser, &secondUser)
 
-	groups, err = ListGroups(db, ByName(engineers.Name))
-	assert.NilError(t, err)
-	assert.Equal(t, 1, len(groups))
+	t.Run("all", func(t *testing.T) {
```

Comment:
Could you elaborate? This is the data layer so no authorization here. 

Don't the other two test cases show that only matching results are returned (because there are groups that do not match the query) ?